### PR TITLE
B2C-7627 - Project setup

### PR DESCRIPTION
**Pull request by pmrotule**
Create Github action to add the Jira ticket to the PR title and add links to the PR description. I only slightly modified the one we already had in a private repository and made it generic enough to be moved to this public repository.

### Summary of changes compared to the previous private action

- Added `jira-account`, `ticket-regex`, `clean-title-regex` inputs and renamed `preview-name` to `preview-link`. This was needed in order to remove the logic specific to On.
- I also added `exception-regex` for dependabot and regex flag inputs
- I added tests for missing required inputs (the new inputs) and providing `clean-title-regex` or not
- General project setup (Typescript, Eslint, Prettier, CI)

### Checklist

- [x] Unit tests are implemented

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    description: Regex to match jira ticket in branch name (i.e. ^ABC-\d+)
+    required: true
+  clean-title-regex:
+    description: Regex used to delete text from PR title
+    required: false
+  preview-link:
+    description: Preview link to add to PR description (i.e. https://preview.example.com)
+    required: false
+runs:
+  using: node16
```
Comment (lneves75) [inline action.yml:22]: Is this the node base image?
If so, we're running CI with node 14.x, shouldn't we use the same node version for the runtime ?

Comment (pmrotule) [reply action.yml:22]: @lneves75 According to [the documentation](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing), we can only specify either `node12` or `node16`. That being said, nothing prevent us from using Node 16 in this repository which is anyway the LTS version.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    const request: Parameters<typeof github.rest.pulls.update>[0] = {
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      pull_number: prNumber,
+    }
+    const prPreviewLine = previewLink ? `**[${PREVIEW_LINK_TEXT}](${previewLink})**\n` : ''
+
+    let ticketLine = ''
+    const headBranch = context.payload.pull_request.head.ref.toUpperCase()
```
Comment (pmrotule) [inline src/index.ts:58]: This transformation of the branch name to uppercase doesn't make sense in a generic scope. We should simply add another input where we can specify if we want it to be case sensible or not (default to case insensible I would say). I need to change it.

I will also add a new input `exception-regex` with a default to `^dependabot\/` in order to allow engineers to push commits to a Dependabot PR.

File: action.yml (hunk 1 above)
Comment (pmrotule) [reply action.yml:36]: So now the project uses Node 16.13.2 👍

**Comment by pmrotule** [comment]
@lneves75 Would you know by any chance why the CI is not running on this PR? For linting and test.

**Review by iuridepaula — APPROVED** [review]
Looks good to me, well done 🎉
If @lneves75 has no other comments, I think it's good to go.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  pull_request_target:
```
Comment (lneves75) [inline .github/workflows/ci.yml:4]: It should be `pull_request`

**Review by lneves75 — APPROVED** [review]
JS review aside 😄 ,  LGTM 👍

**Pull request merged**